### PR TITLE
feat(hub): refine hub http api

### PR DESCRIPTION
**Pull request by kallydev**
Test URL

```text
http://localhost:8080/api/account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum
http://localhost:8080/api/account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum/list/link/following/0
http://localhost:8080/api/account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum/list/backlink?limit=10
```

- [x] Implement business logic (account, linklist, backlinklist)
- [x] Wrap SQL logic
- [x] Database logger

### CODE REVIEW

File: hub/pkg/middleware/instance.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	"net/http"
+
+	"github.com/NaturalSelectionLabs/RSS3-PreGod/hub/pkg/status"
+	"github.com/NaturalSelectionLabs/RSS3-PreGod/hub/pkg/web"
+	"github.com/NaturalSelectionLabs/RSS3-PreGod/shared/pkg/rss3uri"
+	"github.com/gin-gonic/gin"
+)
+
+const (
+	KeyInstance = "instance"
```
Comment (Songkeys) [inline hub/pkg/middleware/instance.go:13]: would it be better to call this `CtxKeyXXX`? and placed in a dir called e.g.`ctx` `constants` to have a overall control of context keys?

File: hub/pkg/router/api/index.go [hunk 2]
```diff
@@ -15,53 +16,43 @@ import (
...
+	if !exists {
 		w := web.Gin{C: c}
-		w.JSONResponse(http.StatusBadRequest, status.INVALID_PARAMS, nil)
+		w.JSONResponse(http.StatusBadRequest, status.CodeInvalidParams, nil)
 
 		return
 	}
 
-	accountInstance, ok := instance.(*rss3uri.PlatformInstance)
+	platformInstance, ok := value.(*rss3uri.PlatformInstance)
```
Comment (kallydev) [inline hub/pkg/router/api/index.go:31]: this is because we may later have to deal with note and asset

File: hub/pkg/middleware/instance.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	Instance string `uri:"instance" binding:"required"`
+}
+
+func Instance() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		request := InstanceUri{}
+		if err := c.ShouldBindUri(&request); err != nil {
+			w := web.Gin{C: c}
+			w.JSONResponse(http.StatusBadRequest, status.CodeInvalidParams, nil)
+			c.Abort()
```
Comment (Songkeys) [inline hub/pkg/middleware/instance.go:26]: if we are not calling `c.Abort()` here, is it going to continue to the handler just like calling a `c.Next()`?

Comment (kallydev) [reply hub/pkg/middleware/instance.go:26]: yes, that will return two response

Comment (kallydev) [reply hub/pkg/middleware/instance.go:26]: image

Comment (Songkeys) [reply hub/pkg/middleware/instance.go:26]: that' weird. i thought `return` would do the aborting. so this's not a classic "onion" model.

File: hub/pkg/router/api/assetlist.go [hunk 4]
```diff
@@ -3,11 +3,15 @@ package api
 import (
```
Comment (Songkeys) [inline hub/pkg/router/api/assetlist.go:3]: i noticed that the file is called `assetlist`, no `get`/`post`. are we going to write post method in the same file?

File: hub/pkg/middleware/instance.go (hunk 1 above)
Comment (kallydev) [reply hub/pkg/middleware/instance.go:13]: i'm not sure if it will be used elsewhere

Comment (Songkeys) [reply hub/pkg/middleware/instance.go:13]: it's not only about this key. we may have some other context keys later. even set some keys in router handler and get them at the end of some middleware.

File: hub/pkg/router/api/assetlist.go (hunk 4 above)
Comment (kallydev) [reply hub/pkg/router/api/assetlist.go:3]: for now yes, i'm starting to understand your previous plan, i'm sorry i did some breaking change before

**Comment by kallydev** [comment]
the note and the asset should be the same kind of item, i will combine them later

**Comment by kallydev** [comment]
http://localhost:8080/api/account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum

```json
{
  "version": "v0.4.0",
  "identifier": "rss3://account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum",
  "date_created": "",
  "date_updated": "",
  "signature": "",
  "profile": {
    "name": "Nya Candy",
    "avatars": [
      "ipfs://QmV6EWdw6Dt4btesnfpGCPDUNfAFHFSoiEsm2AF6YdzYBY"
    ],
    "bio": "🍬 ℭ𝔬𝔯𝔢 𝔓𝔯𝔬𝔠𝔢𝔰𝔰 ℜ𝔢𝔩𝔬𝔞𝔡𝔦𝔫𝔤 ... \nCɾყρƚσ Adventurer\nNya~ 🍭<SITE#candinya.com>",
    "accounts": [
      {
        "identifier": "rss3://account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum"
      }
    ]
  },
  "links": {
    "identifiers": [
      {
        "type": "following",
        "identifier_custom": "rss3://account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum/list/link/following/0",
        "identifier": "rss3://account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum/list/link/following"
      }
    ],
    "identifier_back": "rss3://account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum/list/backlink"
  },
  "items": {
    "notes": {
      "identifier_custom": "rss3://account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum/list/note/0",
      "identifier": "rss3://account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum/list/note"
    },
    "assets": {
      "identifier_custom": "rss3://account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum/list/asset/0",
      "identifier": "rss3://account:0xD3E8ce4841ed658Ec8dcb99B7a74beFC377253EA@ethereum/list/asset"
    }
  }
}
```

**Comment by kallydev** [comment]
wait for #45  to complete

**Pull request merged**